src/placeos-driver/proxy/remote_driver.cr

```diff
@@ -212,8 +214,9 @@ module PlaceOS::Driver::Proxy
     def debug
       module_id = module_id?
       raise Error.new(ErrorCode::ModuleNotFound, "could not find module id", *@error_details) unless module_id
-      client = Core::Client.client(which_core, request_id)
-      client.debug(module_id)
+      Core::Client.client(which_core, request_id) do |client|
+        client.debug(module_id)
```

Comment:
You can try this
```suggestion
        client.debug(module_id) do |message|
          yield message
        end
```

Not sure that it will work due to block-forwarding semantics (explicit block capture may be required), but give it a shot.